scripts/python/assets-automation/generate_assets_report.py

```diff
@@ -0,0 +1,517 @@
+
+        if evaluation == 1:
+            result.packages_using_proxy.append(pkg.replace(sdk_path + os.sep, ""))
+        elif evaluation == 2:
+            result.packages_using_proxy.append(pkg.replace(sdk_path + os.sep, ""))
+            result.packages_using_external.append(pkg.replace(sdk_path + os.sep, ""))
+
+    print("done.")
+
+    return result
+
+
+def evaluate_net_package(csproj_path: str) -> int:
+    evaluation = 0
+    found_recorded_testcase = False
+    possible_test_directory = os.path.join(os.path.dirname(csproj_path), "..", "tests")
+    possible_project_assets_json = os.path.join(os.path.dirname(csproj_path), "..", "assets.json")
+    possible_solution_assets_json = os.path.join(os.path.dirname(csproj_path), "..", "..", "assets.json")
+    session_records = os.path.join(possible_test_directory, "SessionRecords")
+    package_name = os.path.splitext(os.path.basename(csproj_path))[0]
+
+    if not os.path.exists(possible_test_directory):
+        return 0
+
+    # for Azure.*, only examine packages with recorded tests. EG with existing SessionRecords or an existing assets.json
+    if not os.path.exists(session_records) and not (
+        os.path.exists(possible_project_assets_json) or os.path.exists(possible_solution_assets_json)
+    ):
+        return 0
+
+    # For mgmt, you should find a reference to ManagementRecordedTestBase in projects using test proxy:
+    #  https://grep.app/search?q=managementrecordedtestbase&filter[repo][0]=Azure/azure-sdk-for-net&filter[path][0]=sdk/
+    # For data plane, you should find RecordedTestBase:
+    #  https://grep.app/search?q=recordedtestbase&filter[repo][0]=Azure/azure-sdk-for-net&filter[path][0]=sdk/
+
+    find = "RecordedTestBase"
+    if "ResourceManager" in package_name:
+        find = "ManagementRecordedTestBase"
+
+    test_files = glob.glob(os.path.join(possible_test_directory, "**", "*.cs"), recursive=True)
+
+    for testfile in test_files:
+        try:
+            with open(testfile, "r", encoding="utf-8") as f:
+                content = f.read()
+
+                if find in content:
+                    evaluation = 1
+        except:
+            pass
+
+    if os.path.exists(possible_project_assets_json) or os.path.exists(possible_solution_assets_json):
+        evaluation = 2
+
+    return evaluation
+
+
+def net_trim_path(solution_path: str) -> str:
+    return os.path.splitext(os.path.basename(solution_path))[0]
+
+
+def generate_net_report() -> ScanResult:
+    language = "net"
+    result = ScanResult("." + language.upper())
+    repo = get_repo(language)
+
+    print(f"Evaluating repo for {language} @ {repo}", end="...")
+
+    #                                                     <service>
+    #                                                         |<package>
+    #                                                         |    |
+    all_azure_projects = glob.glob(os.path.join(repo, "sdk", "*", "*", "src", "*Azure.*.csproj"), recursive=True)
```

Comment:
I assume the exact counts aren't critical for this report but if we did need exact counts we should try to use the Get-AllPackageInfoFromRepo functions we have in each language-settings.ps1 to discover the packages.

Reply:
I see where you're coming from but given how little of this script is actually _discovering_ the package, I don't necessarily see a point in debugging getting that working.